bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -199,9 +199,8 @@ func (r *KubeadmConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, err
 	}
 
-	if annotations.IsPaused(cluster, config) {
-		log.Info("Reconciliation is paused for this object")
-		return ctrl.Result{}, nil
+	if isPaused, conditionChanged, err := paused.EnsurePausedCondition(ctx, r.Client, cluster, config); err != nil || isPaused || conditionChanged {
```

Comment:
Should I add such a comment everywhere?
```suggestion
	// Return early if the object or Cluster is paused or the condition needs to get patched.
	if isPaused, conditionChanged, err := paused.EnsurePausedCondition(ctx, r.Client, cluster, config); err != nil || isPaused || conditionChanged {
```

Reply:
IMO the code read well also without a comment, but up to you